universal installer 0.3

fatz opened the pull request:
## Jira Ticket

https://jira.d2iq.com/browse/D2IQ-70173

## Description of changes being made
move universal installer 0.2 docs into deprecated state and add upgrade guides
update universal installer docs to be used with version 0.3

### Changes are made to 2.1 and 2.2 tree of docs:
- added universal installer 0.3 templates
- changed versioning in TF examples to use to model variable for DC/OS version
- added upgrade guide to universal installer 0.2 docs. 
- ported these changes into 2.2 folder

## Checklist
- [ ] Change all affected versions, if applicable (e.g. 1.13, 2.0, 2.1).
- [ ] Test all commands and procedures, if applicable.
- [ ] Create your PR against `staging`, not `master`. 
- [ ] Provide an estimated date for deploying the doc change. Note: Improvements or fixes can be merged ASAP. 
- [ ] Add [redirects](https://github.com/mesosphere/dcos-docs-site/wiki/Redirects) if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> ie `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/wiki/Contributing) for more information.

fatz commented:
Hey @cmastr AWS is done so far. I still need to add GCP and Azure. I'm out tomorrow so will have a look on that next week. Could you maybe have a look at my AWS related changes already?

cmastr commented:
> Hey @cmastr AWS is done so far. I still need to add GCP and Azure. I'm out tomorrow so will have a look on that next week. Could you maybe have a look at my AWS related changes already?

Yes will try to take a look today.

cmastr commented on pages/mesosphere/dcos/2.1/installing/evaluation/index.md:
```diff
@@ -4,7 +4,7 @@ navigationTitle:  Cloud Installation
 menuWeight: 10
 excerpt: Guide to Installing DC/OS on cloud environments using the Mesosphere Universal Installer
-model: /mesosphere/dcos/2.1/data.yml
+model: /mesosphere/dcos/2.2/data.yml
```
Is this change for 2.2? If so, we need to move all these changes to the 2.2 branch, which has already been created on staging. Otherwise, if this change is for 2.1, we should remove this model name change.

fatz replied:
i think its for both... just needed to start somewhere

cmastr commented:
@janisz I need some clarification on what we're doing here. Is this a change for 2.2? Why would we rename files? We need to still have these files for 2.1, then have a duplicate for 2.2 which in my mind means copying them over then renaming. If you just rename, that removes them from the previous version of the docs.

cmastr commented:
Ok, got it now. Will stop bothering you now that I know it's still in progress. thanks!

fatz commented:
Ok @cmastr got most stuff working now. There is an minor issue with GCP remote region which I'll try to fix tomorrow morning

cneth commented on pages/mesosphere/dcos/2.1/installing/evaluation/universal-installer-0.2/aws/upgrade/index.md:
````diff
@@ -0,0 +1,161 @@
+```
+$ terraform apply
+```
+The Plan will show create statements for `aws_lb_target_group_attachment` this is completely ok. Make sure there is __no__ destroy in this statement. With the create statements terraform will realise that the attachment already exists and places this information in its state.
````
Suggested rewording:

"The resulting plan should show create actions for the `aws_lb_target_group_attachment` module, but no others.  There should be __no__ destroy actions, if there are, then you should answer no to the prompt and review the above steps again (??? what do you do here if you see a destroy).   

After you enter yes to proceed, Terraform will not actually create new attachments but will detect the existing attachments and add them to its state."

cneth commented on pages/mesosphere/dcos/2.1/installing/evaluation/universal-installer-0.2/aws/upgrade/index.md:
````diff
@@ -0,0 +1,161 @@
+$ terraform apply
+```
+The Plan will show create statements for `aws_lb_target_group_attachment` this is completely ok. Make sure there is __no__ destroy in this statement. With the create statements terraform will realise that the attachment already exists and places this information in its state.
+After this we're done.
````
Your Terraform installation is now completely upgraded to v0.3.

cneth commented on pages/mesosphere/dcos/2.1/installing/evaluation/universal-installer-0.2/azure/upgrade/index.md:
```diff
@@ -0,0 +1,191 @@
+
+
+# Upgrade Universal Installer 0.2 to 0.3
+This guide helps you upgrading Universal Installer 0.2 to 0.3 to support terraform 0.12
+With this guide we assume you’re running a DC/OS cluster similar to our example in the docs. More complex setup might need more care. But the here mentioned changes should also apply. Before doing this on your production cluster you should setup a test cluster in a similar way to make sure changes won’t render your cluster unresponsive.
+
+## Preparation
+Ideally you should use tfenv to switch between terraform versions but you can also do it by yourself. so tfenv use 0.12.25 means you need to replace your 0.11 terraform version with 0.12.25
```
Please see the AWS page for suggesting rewordings for these paragraphs.

cneth commented on pages/mesosphere/dcos/2.1/installing/evaluation/universal-installer-0.2/azure/upgrade/index.md:
````diff
@@ -0,0 +1,191 @@
+$ terraform apply
+```
+
+## Translate terraform 0.11 `main.tf` into 0.12
+Now we switch to terraform 0.12.25 which offers us an option to translate terraform 0.11 code into terraform 0.12 code
````
See AWS guid for suggestions on how to reword this.

cneth commented on pages/mesosphere/dcos/2.1/installing/evaluation/universal-installer-0.2/azure/upgrade/index.md:
````diff
@@ -0,0 +1,191 @@
+First we upgrade our modules to Universal Installer 0.3 ( version change from above)
+
+```
+$ terraform init -upgrade
````
Note wording suggestions from AWS as well as removing `$`

cneth commented on pages/mesosphere/dcos/2.1/installing/evaluation/universal-installer-0.2/gcp/upgrade/index.md:
```diff
@@ -0,0 +1,171 @@
+---
+
+
+# Upgrade Universal Installer 0.2 to 0.3
+This guide helps you upgrading Universal Installer 0.2 to 0.3 to support terraform 0.12
+With this guide we assume you’re running a DC/OS cluster similar to our example in the docs. More complex setup might need more care. But the here mentioned changes should also apply. Before doing this on your production cluster you should setup a test cluster in a similar way to make sure changes won’t render your cluster unresponsive.
+
+## Preparation
+Ideally you should use tfenv to switch between terraform versions but you can also do it by yourself. so tfenv use 0.12.25 means you need to replace your 0.11 terraform version with 0.12.25
+We assume you’re using a clusters main.tf similar to this:
+
```
Please see the aws pages for suggested rewordings of these common paragraphs.

cneth commented on pages/mesosphere/dcos/2.1/installing/evaluation/universal-installer-0.2/index.md:
```diff
@@ -0,0 +1,22 @@
+---
+
+# Deprecation Notice
+Universal Installer 0.2 is based on Terraform 0.11 which is EOL and does not get any further updates. Furthermore Terraform providers might get incompatible with this version of Terraform. We strongly suggest to update to Universal Installer 0.3 using the provided Upgrade guides
```
Suggested rewording:

"Universal Installer 0.2 is based on Terraform v0.11, which is EOL and no longer updated.  Because it is EOL, it is possible that some non-DC/OS Terraform providers might eventually become incompatible with v0.11 of Terraform. Therefore, we strongly suggest that you do not create any new clusters using the Universal Installer 0.2 modules, and that you upgrade any existing 0.2 based cluster to the 0.3 version of the modules using the upgrade guides listed below."

fatz commented on pages/mesosphere/dcos/install-include-0.3/all-install-terraform.tmpl:
````diff
@@ -0,0 +1,17 @@
+    brew unlink terraform || true
+    brew install tfenv
+    tfenv install 0.12.25
+    ```
````
indeed

fatz commented:
the thing is that `tfenv install v0.12.25` doesn't work. Same vor HCL version constraints which are plain semantic versioning.

I'd really like to avoid `vx.x.x` for now

fatz commented:
@cneth So i've used `jq-1.6` maybe thats mandatory can you check if you have the same version?

Do you still have the error you got somewhere? Would be great to find out what went wrong here

vinodkone commented on pages/mesosphere/dcos/2.2/installing/index.md:
```diff
@@ -12,18 +12,18 @@ The Mesosphere Universal Installer is now the supported installation method for
 ##### Jump to the getting started guide for your cloud provider:
 
-#### [Amazon Web Services](/mesosphere/dcos/2.2/installing/evaluation/aws/)
+#### [Amazon Web Services](/mesosphere/dcos/2.1/installing/evaluation/aws/)
```
why is this pointing to 2.1 page?

vinodkone commented on pages/mesosphere/dcos/2.2/installing/installation-faq/index.md:
```diff
@@ -75,4 +75,4 @@ The IAM database is restored from the backup file and the cluster is operational
 ## Q. How do I back up ZooKeeper?
 
-To back up ZooKeeper, follow the guide to back up and restore the [DC/OS ZooKeeper State](/mesosphere/dcos/2.2/administering-clusters/backup-and-restore/backup-restore-cli/#zookeeper).
+To back up ZooKeeper, follow the guide to back up and restore the [DC/OS ZooKeeper State](/mesosphere/dcos/2.1/administering-clusters/backup-and-restore/backup-restore-cli/#zookeeper).
```
ditto.

fatz replied:
well we can copy it to other versions aswell... I was thinking if we drop out the universal installer 0.2 docs in future releases and just refer back to the one when it was introduced...

@cmastr any thought about this

fatz replied:
ah guess this is a copy paste issue...

vinodkone commented on pages/mesosphere/dcos/2.2/installing/production/advanced-configuration/configuring-gpu-nodes/index.md:
```diff
@@ -19,18 +19,18 @@ GPUs must be enabled during DC/OS installation. Follow the instructions below to
 
 1.  Install the [NVIDIA Management Library (NVML)](https://developer.nvidia.com/nvidia-management-library-nvml) on each node of your cluster that has GPUs. The minimum required NVIDIA driver version is 340.29. For detailed installation instructions, see the [Mesos GPU support documentation](http://mesos.apache.org/documentation/latest/gpu-support/#external-dependencies).
-1.  Install DC/OS using the [custom advanced installation instructions](/mesosphere/dcos/2.2/installing/production/deploying-dcos/installation/). Here are the GPU-specific configuration parameters:
+1.  Install DC/OS using the [custom advanced installation instructions](/mesosphere/dcos/2.1/installing/production/deploying-dcos/installation/). Here are the GPU-specific configuration parameters:
```
ditto.

fatz commented:
@vinodkone I did a search and replace in 2.2 folder for whatever was doing wrong 2.1 links. I've added the templating model to those files and replaced `2.1` with `{{ model.folder_version }}` which should prevent this from happening in the future.

its in 7a95cbf so if you don't wanna have it just revert

cmastr commented:
@fatz did you already implement comments from Craig? That needs to be done before I can merge this.

cmastr commented:
going to start merging this. If any outstanding comments, open a new PR to address.
